pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/predicate/RangeRealtimeDictionaryPredicateEvaluator.java

```diff
@@ -69,6 +69,7 @@ public RangeRealtimeDictionaryPredicateEvaluator(RangePredicate predicate, Mutab
     dictIdSet = new IntOpenHashSet(ids.size());
     for (int i = 0; i < matchingIds.length; i++) {
       matchingIds[i] = ids.get(i);
+      dictIdSet.add(ids.get(i));
```

Comment:
Good catch. Wonderful if you can add a unit test for this. Thanks, much appreciated.